src/callbacks_step/save_solution.jl

```diff
@@ -85,30 +86,34 @@ end
 
 
 function SaveSolutionCallback(; interval::Integer=0,
-                                dt=0.0,
+                                dt=nothing,
                                 save_initial_solution=true,
                                 save_final_solution=true,
                                 output_directory="out",
                                 solution_variables=cons2prim)
 
-  if dt > 0
-    interval = dt
+  if !isnothing(dt) && interval == 0
+    interval_or_dt = dt
+  else # isnothing(dt) && interval >= 0
+    interval_or_dt = interval
   end
```

Comment:
```suggestion
  if !isnothing(dt)
    interval_or_dt = dt
  else # isnothing(dt)
    interval_or_dt = interval
  end
```
The idea was to use a logic depending only on the types. Could you please check whether this version is type stable (using `Test.@inferred` and/or `@code_warntype`)?

Reply:
Okay, I misunderstood the strategy of the logic. I will update and test this variant for type stability.

Reply:
I used `@code_warntype` calling the `SaveSolutionCallback` function with either `interval` or `dt` inputs. I do not get any non-concrete types (i.e. no red text) which I assume is a good thing.

Reply:
Yes, that's great!

Reply:
Before this if condition, it would imho also be good to check something like
```julia
if interval > 0 && !isnothing(dt)
  error("You can either set the number of steps between output (using `interval`) or the time between outputs (using `dt`) but not both simultaneously")
end
```
(or something shorter 😬) to prevent user surprises if they set both and expect it to somehow work.